requirement-generator/fetcher.ts

```diff
@@ -85,13 +85,15 @@ const getAllCoursesInSemester = async <T extends keyof Course>(
   return courses;
 };
 
+type AllCourses<T extends keyof Course> = { [semester: string]: readonly Pick<Course, T>[] };
+
 const getAllCourses = async <T extends keyof Course>(
   courseFieldFilter: CourseFieldFilter<T>,
   coolingTimeMs: number = 50,
   doPrintDebuggingInfo: boolean = true
-): Promise<readonly Pick<Course, T>[]> => {
+): Promise<AllCourses<T>> => {
   const startTime = new Date().getTime();
```

Comment:
Is this ever used?

Reply:
This is used to generate the courses json

Reply:
Apologies, I meant to specify `startTime`

Reply:
But nvm, I see that it's used for console logging